modules/porous_flow/src/actions/PorousFlowSinglePhaseBase.C

```diff
@@ -214,6 +214,8 @@ PorousFlowSinglePhaseBase::addKernels()
       if (_coord_system == Moose::COORD_RZ)
         kernel_type = "StressDivergenceRZTensors";
       InputParameters params = _factory.getValidParams(kernel_type);
+      if (_subdomain_names_set)
+      params.set<std::vector<SubdomainName>>("block") = _subdomain_names;
```

Comment:
I should have mentioned, you dont need to do this for kernels as they are automatically block-restricted to the nonlinear variable block-restriction

only variables, materials, UOs, ...